.tekton/collector-component-pipeline.yaml

```diff
@@ -0,0 +1,401 @@
+    params:
+    - name: url
+      value: $(params.git-url)
+    - name: revision
+      value: $(params.revision)
+    - name: submodules
+      value: $(params.clone-submodules)
+    - name: depth
+      value: "$(params.clone-depth)"
+    - name: fetchTags
+      value: "$(params.clone-fetch-tags)"
+    runAfter:
+    - init
+    taskRef:
+      params:
+      - name: name
+        value: git-clone
+      - name: bundle
+        value: quay.io/redhat-appstudio-tekton-catalog/task-git-clone:0.1@sha256:30709df067659a407968154fd39e99763823d8ecfc6b5cd00a55b68818ec94ba
+      - name: kind
+        value: task
+      resolver: bundles
+    when:
+    - input: $(tasks.init.results.build)
+      operator: in
+      values: [ "true" ]
+    workspaces:
+    - name: output
+      workspace: workspace
+    - name: basic-auth
+      workspace: git-auth
+
+  - name: determine-image-tag
+    runAfter:
+    # This task must run on a freshly cloned repository to prevent seeing any changes from other tasks.
+    - clone-repository
+    taskRef:
+      name: determine-image-tag
+    workspaces:
+    - name: source
+      workspace: workspace
+
+  - name: prepare-rhel-rpm-subscriptions
+    runAfter:
+    - determine-image-tag
+    workspaces:
+    - name: source
+      workspace: workspace
+    - name: subscription-manager-activation-key
+      workspace: subscription-manager-activation-key
+    taskSpec:
+      steps:
+      # TODO(ROX-20651): use content sets instead of subscription manager for access to RHEL RPMs once available.
+      - name: smuggle-activation-key
+        image: registry.access.redhat.com/ubi8/ubi:latest
+        script: exec "$(workspaces.source.path)/source/.konflux/scripts/subscription-manager-bro.sh" smuggle
+
+  - name: prefetch-dependencies
+    params:
+    - name: input
+      value: $(params.prefetch-input)
+    runAfter:
+    - determine-image-tag
+    taskRef:
+      params:
+      - name: name
+        value: prefetch-dependencies
+      - name: bundle
+        value: quay.io/redhat-appstudio-tekton-catalog/task-prefetch-dependencies:0.1@sha256:c6fdbf404dc61bf8cf8bec5fc4d7fb15f37ba62f1684de0c68bfbad5723c0052
+      - name: kind
+        value: task
+      resolver: bundles
+    workspaces:
+    - name: source
+      workspace: workspace
+
+  - name: build-container
+    params:
+    - name: IMAGE
+      value: $(params.output-image-repo):$(tasks.determine-image-tag.results.image-tag)$(params.output-tag-suffix)
+    - name: DOCKERFILE
+      value: $(params.dockerfile)
+    - name: CONTEXT
+      value: $(params.path-context)
+    - name: HERMETIC
+      value: $(params.hermetic)
+    - name: PREFETCH_INPUT
+      value: $(params.prefetch-input)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.image-expires-after)
+    - name: COMMIT_SHA
+      value: $(tasks.clone-repository.results.commit)
+    - name: TARGET_STAGE
+      value: $(params.build-target-stage)
+    runAfter:
+    - prefetch-dependencies
```

Comment:
This seems to be slightly different from the original version, where `prefetch-dependencies` was run after `prepare-rhel-rpm-subscriptions`. Just out of curiosity, is the order here important?

Reply:
The order matters in general but not here.

Both `prefetch-dependencies` and `prepare-rhel-rpm-subscriptions` (as well as earlier `clone-repository` and `determine-image-tag`) are prerequisites for `build-container`. They were put sequentially because, well, we could. They can also be put parallel because they don't have a dependency on each other.

In fact, in `.tekton/collector-(pull-request|push).yaml` they were already in parallel and this caused no issue.
`.tekton/collector-slim-(pull-request|push).yaml` were simply not made consistent.